src/operator/tensor/cast_storage-inl.cuh

```diff
@@ -188,6 +188,15 @@ struct FillRspValsKernel {
   }
 };
 
+template<typename xpu, int ndim, typename DType>
+inline mshadow::Tensor<xpu, ndim, DType> AllocateTempDataForCast(const OpContext& op_ctx,
```

Comment:
I intention ally named it after the Cast usage in order to discourage it from being used in a general way to circumvent the proper temp resource allocation mechanism.